core/infra/config/peer_service_with_json.cpp

```diff
@@ -180,28 +181,22 @@ bool PeerServiceConfig::isExistPublicKey( const std::string &publicKey ) {
   return findPeerPublicKey( std::move(publicKey) ) != peerList.end();
 }
 
-std::vector<peer::Node>::iterator PeerServiceConfig::findPeerIP( const std::string &ip ) {
+Nodes::iterator PeerServiceConfig::findPeerIP( const std::string &ip ) {
   return std::find_if( peerList.begin(), peerList.end(),
-                       [&ip]( const peer::Node& p ) { return p.getIP() == ip; } );
+                       [&ip]( const auto& p ) { return p->getIP() == ip; } );
 }
 
-std::vector<peer::Node>::iterator PeerServiceConfig::findPeerPublicKey( const std::string &publicKey ) {
+Nodes::iterator PeerServiceConfig::findPeerPublicKey( const std::string &publicKey ) {
   return std::find_if( peerList.begin(), peerList.end(),
-                       [&publicKey]( const peer::Node& p ) { return p.getPublicKey() == publicKey; } );
+                       [&publicKey]( const auto& p ) { return p->getPublicKey() == publicKey; } );
 }
 
-
-std::vector<std::unique_ptr<peer::Node>> PeerServiceConfig::getPeerList() {
+Nodes PeerServiceConfig::getPeerList() {
   initialziePeerList_from_json();
 
-  std::vector<std::unique_ptr<peer::Node>> nodes;
-  for(const auto &node : peerList) {
-    if(node.isOK()) {
-        nodes.push_back(std::make_unique<peer::Node>(node.getIP(),
-                                                     node.getPublicKey(),
-                                                     node.getTrustScore()));
-    }
-  }
+  Nodes nodes(peerList.size());
+  std::copy_if(peerList.begin(), peerList.end(), nodes.begin(),
+               [](const auto &node) { return (node)->isOK(); });
```

Comment:
Thanks, but I don't know whether getPeerList() allows peerList's destructive change or not.